src/basic_recipes/waterfall.jl

```diff
@@ -0,0 +1,123 @@
+"""
+    waterfall(x, y; kwargs...)
+
+Plots a [waterfall chart](https://en.wikipedia.org/wiki/Waterfall_chart) to visualize individual
+positive and negative components that add up to a net result as a barplot with stacked bars next
+to each other.
+
+## Attributes
+$(ATTRIBUTES)
+
+Furthermore the same attributes as for `barplot` are supported.
+"""
+@recipe(Waterfall, x, y) do scene
+    return Attributes(;
+        dodge=automatic,
+        n_dodge=automatic,
+        gap=0.2,
+        dodge_gap=0.03,
+        width=automatic,
+        cycle=[:color => :patchcolor],
+        stack=automatic,
+        show_direction=false,
+        marker_pos=:utriangle,
+        marker_neg=:dtriangle,
+        direction_color=theme(scene, :backgroundcolor),
+        show_final=false,
+        final_color=plot_color(:grey90, 0.5),
+        final_gap=automatic,
+        final_dodge_gap=0,
+    )
+end
+
+conversion_trait(::Type{<:Waterfall}) = PointBased()
+
+function Makie.plot!(p::Waterfall)
+    function stack_bars(xy, dodge, stack)
+        x, y = first.(xy), last.(xy)
+        if stack === automatic
+            stack = dodge === automatic ? :x : :dodge
+        end
+        i_dodge = dodge === automatic ? ones(Int, length(x)) : dodge
+        i_stack, i_group = stack === :dodge ? (i_dodge, x) : (x, i_dodge)
+        xy = similar(xy)
+        fillto = similar(x)
+        final = similar(xy)
+        groupby = StructArray(; grp=i_group)
+        for (grp, inds) in StructArrays.finduniquesorted(groupby)
+            fromto = stack_from_to_final(i_stack[inds], y[inds])
+            fillto[inds] .= fromto.from
+            xy[inds] .= Point2f.(x[inds], fromto.to)
+            final[inds] .= Point2f.(x[inds], fromto.final)
+        end
+        return (xy=xy, fillto=fillto, final=final)
+    end
+
+    fromto = lift(stack_bars, p[1], p.dodge, p.stack)
+
+    if p.show_final[]
+        final_gap = p.final_gap[] === automatic ? p.dodge[] == automatic ? 0 : p.gap : p.final_gap
+        barplot!(
+            p,
+            fromto[].final;
```

Comment:
You should turn all the `fromto[]`/`marker[]`/etc into: `lift(x-> x.xy, fromto)`

Reply:
Thanks for the hint! Does this also apply to `p.show_final[]`, `p.final_gap[]` etc?

Reply:
Yes, but there you should be able to just remove the []

Reply:
Hm, but if I remove `[]` e.g. in L58, I get
```julia
ERROR: TypeError: non-boolean (Observable{Any}) used in boolean context
```